### ISSUE TITLE
after setting change for enterprise+ customers project_key is a required input for local repos

### ISSUE THREAD

**Issue by wwsean08** [jfrog/terraform-provider-artifactory#647]
**Describe the bug**
Last week our SaaS enterprise+ server got upgraded from `7.47.12` to `7.50.7` and according to support a new settings change was rolled out to all enterprise+ customers to enable projects.  Once that happened our automation for creating/maintaining jfrog repos broke where where it was trying to remove it from the default project.

Plan output (note the entire log is way too long this is just an example of one of the repos that from the logs:
```
  # artifactory_local_generic_repository.generic["REDACTED"] will be updated in-place
  ~ resource "artifactory_local_generic_repository" "generic" {
        id                       = "REDACTED-generic"
      - project_key              = "default" -> null
        # (12 unchanged attributes hidden)
    }
```

The error:
```
│ Error: 
│ 404 DELETE https://REDACTED.jfrog.io/access/api/v1/projects/_/attach/repositories/REDACTED-generic
│ {
│   "errors" : [ {
│     "code" : "NOT_FOUND",
│     "message" : "Could not find resource `REDACTED-generic`; type `repo` in Projects configuration"
│   } ]
│ }
│ 
│   with artifactory_local_generic_repository.generic["REDACTED"],
│   on artifactory.tf line 54, in resource "artifactory_local_generic_repository" "generic":
│   54: resource "artifactory_local_generic_repository" "generic" {
```

The workaround that I came up with support was to add the `project_key` attribute to the `artifactory_local_generic_repository` and `artifactory_local_docker_v2_repository` objects.

**Requirements for and issue**
- [x] A description of the bug
- [X] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue). **If this is not supplied, this issue will likely be closed without any effort expended.**
- [X] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`) - enterprise+ SaaS server at `7.50.7`
- [X] Your version of terraform - `v1.3.8`
- [X] Your version of terraform provider - tested on `6.24.2` and `6.26.1` and happens regardless

**Expected behavior**
Either project_key should be updated to handle the way entperprise+ SaaS servers work now (preferably), or be made a required field with the relevant updates to documentation.

**Additional context**
minimal reproduction (all you need to do is set the server/token via environment variables):
```hcl
terraform {
  required_providers {
    artifactory = {
      source  = "jfrog/artifactory"
      version = "~> 6.26.0"
    }
  }
}

resource "artifactory_local_generic_repository" "generic" {
  key         = "test-sean-smith-generic"
}
```

Note you will need to apply this once then the second time you will get:

```
artifactory_local_generic_repository.generic: Refreshing state... [id=test-sean-smith-generic]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # artifactory_local_generic_repository.generic will be updated in-place
  ~ resource "artifactory_local_generic_repository" "generic" {
        id                       = "test-sean-smith-generic"
      - project_key              = "default" -> null
        # (11 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.```

**Comment by mcaulifn** [comment]
@alexhung @danielmkn Is there an update on this?

**Comment by alexhung** [comment]
@mcaulifn We've triage it but have not put into our work schedules. Let us know if this is urgent and we can bump this up the queue.

**Comment by alexhung** [comment]
We've moved the ticket into our current sprint. We'll start work on it next week.

**Comment by alexhung** [comment]
@wwsean08 @mcaulifn We will be releasing a major version bump for this, to v7.x.x. Reason being that we don't have a solution that is backward compatible and works for all the scenarios (pre 7.50.x, post 7.50.x, importing, with `project_key` set/not set, etc.).

We are planning to maintain both v6 and v7 simultaneously for a while until/if we find a solution that work for everyone.

**Comment by danielmkn** [comment]
@wwsean08, we released v7.0.0 with the default value for that attribute, so you won't have a state drift in your config.
